src/integer/z/arithmetic/add.rs

```diff
@@ -140,6 +144,51 @@ impl Add<&Zq> for &Z {
 arithmetic_trait_borrowed_to_owned!(Add, add, Z, Zq, Zq);
 arithmetic_trait_mixed_borrowed_owned!(Add, add, Z, Zq, Zq);
 
+impl Add<&PolyOverZ> for &Z {
+    type Output = PolyOverZ;
+    /// Implements the [`Add`] trait for [`Z`] and [`PolyOverZ`] values.
+    /// [`Add`] is implemented for any combination of owned and borrowed values.
+    ///
+    /// Parameters:
+    ///  - `other`: specifies the polynomial to add to `self`
+    ///
+    /// Returns the sum of both as a [`PolyOverZ`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer::Z;
+    /// use std::str::FromStr;
+    ///
+    /// let a: Z = Z::from(42);
+    /// let b: PolyOverZ = PolyOverZ::from_str("4  5 1 2 3").unwrap();
+    ///
+    /// let c: PolyOverZ = &a + &b;
+    /// let d: PolyOverZ = a + b;
+    /// let e: PolyOverZ = &Z::from(42) + d;
+    /// let f: PolyOverZ = Z::from(42) + &e;
+    /// ```
```

Comment:
maybe add an `assert_eq!(...)` where the reader can see the actual addition
```suggestion
    /// ```
    /// use qfall_math::integer::PolyOverZ;
    /// use qfall_math::integer::Z;
    /// use std::str::FromStr;
    ///
    /// let a: Z = Z::from(42);
    /// let b: PolyOverZ = PolyOverZ::from_str("4  5 1 2 3").unwrap();
    ///
    /// let c: PolyOverZ = &a + &b;
    /// let d: PolyOverZ = a + b;
    /// let e: PolyOverZ = &Z::from(42) + d;
    /// let f: PolyOverZ = Z::from(42) + &e;
    /// 
    /// assert_eq!(PolyOverZ::from_str!("4  47 1 2 3").unwrap(), a+b)
    /// ```
```
Maybe choose some clever way so the expected behaviour is shown and not just the available types

Reply:
in alinement with the other tests I will keep the current implementation